generate_workspace/src/test/java/com/google/devtools/build/workspace/maven/VersionResolverTest.java

```diff
@@ -0,0 +1,116 @@
+package com.google.devtools.build.workspace.maven;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link VersionResolver}.
+ */
+@RunWith(JUnit4.class)
+public class VersionResolverTest {
+
+  /**
+   * Ensures that an exception is thrown if there is a version range resolution exception.
+   * This occurs when aether is unable to resolve the version.
+   */
+  @Test
+  public void failsOnResolutionException() {
+    Aether aether = Mockito.mock(Aether.class);
+    try {
+      Mockito.when(aether.requestVersionRange(any()))
+          .thenThrow(new VersionRangeResolutionException(any()));
+
+      VersionResolver resolver = new VersionResolver(aether);
+      resolver.resolveVersion("something", "something", "1.0");
+      fail();
+    } catch (VersionRangeResolutionException e) {
+      // This should be caught by the VersionResolver
+      fail();
+    } catch (ArtifactBuilder.InvalidArtifactCoordinateException e) {
+      String expected =
+          "Unable to find a version for something:something:1.0 due to Failed to resolve version range";
+      assertThat(e.getMessage()).isEqualTo(expected);
+    }
+  }
+
+  /**
+   * Ensures that an exception is thrown if there is an invalid version range. An invalid version range
+   * is one which is either (1) equal to null or (2) returns null when asked for highest version.
+   */
+  @Test
+  public void failsOnInvalidVersionRange() {
+    Aether aether = Mockito.mock(Aether.class);
+
+    try {
+      // Using `anyRangeResult()` will ensure that rangeResult.highestVersion() == null.
```

Comment:
Out of date comment. 

Also, use https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests here and above.